domain/consensus/factory.go

```diff
@@ -51,15 +51,21 @@ type Factory interface {
 		externalapi.Consensus, error)
 	NewTestConsensus(dagParams *dagconfig.Params, isArchivalNode bool, testName string) (
 		tc testapi.TestConsensus, teardown func(keepDataDir bool), err error)
-	NewTestConsensusWithDataDir(dagParams *dagconfig.Params, dataDir string, isArchivalNode bool) (
-		tc testapi.TestConsensus, teardown func(keepDataDir bool), err error)
+
+	OverrideDataDir(dataDir string)
+	OverrideGHOSTDAGManager(ghostdagConstructor GHOSTDAGConstructor)
```

Comment:
These should probably be called something like SetTestDataDir and SetTestGHOSTDAGManager, since they apply only to the test consensus.

Reply:
Yeah I thought about Set vs Override, I went with override but Set also works.